src/http_client.rs

```diff
@@ -24,6 +24,9 @@ struct InternalResp {
 type InternalResponseResult = Result<InternalResp, String>;
 type FutureInternalResponse = oneshot::Receiver<InternalResponseResult>;
 
+pub type Response<T> = Result<T, String>;
+pub type FutureResponse<T> = Box<Future<Item = Response<T>, Error = ()>>;
```

Comment:
Being totally unfamiliar with this and looking at this, does it make sense to be

`Box<Future<Item = T, Error = String>>`

Reply:
yeah, it does look like it should be that. maybe can get it there eventually, but i think it maybe can't because i'm using `String` for the error type...

Reply:
There's also `Box<Error>` as the generic just-let-me-return-whatever error type? This is okay too though.

Reply:
I guess that doesn't help if `Error` is supposed to be something that implements the `Error` trait. I'm not sure if `Box<Error>` does or not.